rest/importuserxattrtest/import_test.go

```diff
@@ -0,0 +1,318 @@
+
+	err = rt.WaitForCondition(func() bool {
+		return rt.GetDatabase().DbStats.Database().Crc32MatchCount.Value() == 1
+	})
+	assert.NoError(t, err)
+
+	var syncData2 db.SyncData
+	_, err = dataStore.GetXattr(ctx, docKey, base.SyncXattrName, &syncData2)
+	assert.NoError(t, err)
+
+	assert.Equal(t, syncData.Crc32c, syncData2.Crc32c)
+	assert.Equal(t, syncData.Crc32cUserXattr, syncData2.Crc32cUserXattr)
+	assert.Equal(t, int64(2), rt.GetDatabase().DbStats.Database().SyncFunctionCount.Value())
+	assert.Equal(t, int64(1), rt.GetDatabase().DbStats.SharedBucketImport().ImportCount.Value())
+
+	// Update body but same value and ensure it isn't imported again (crc32 hash should match)
+	err = dataStore.Set(docKey, 0, nil, map[string]interface{}{})
+	assert.NoError(t, err)
+
+	err = rt.WaitForCondition(func() bool {
+		return rt.GetDatabase().DbStats.Database().Crc32MatchCount.Value() == 2
+	})
+	assert.NoError(t, err)
+
+	var syncData3 db.SyncData
+	_, err = dataStore.GetXattr(ctx, docKey, base.SyncXattrName, &syncData3)
+	assert.NoError(t, err)
+
+	assert.Equal(t, syncData2.Crc32c, syncData3.Crc32c)
+	assert.Equal(t, syncData2.Crc32cUserXattr, syncData3.Crc32cUserXattr)
+	assert.Equal(t, int64(2), rt.GetDatabase().DbStats.Database().SyncFunctionCount.Value())
+	assert.Equal(t, int64(1), rt.GetDatabase().DbStats.SharedBucketImport().ImportCount.Value())
+
+	// Update body and ensure import occurs
+	updateVal := []byte(`{"prop":"val"}`)
+	err = dataStore.Set(docKey, 0, nil, updateVal)
+	assert.NoError(t, err)
+
+	err = rt.WaitForCondition(func() bool {
+		return rt.GetDatabase().DbStats.SharedBucketImport().ImportCount.Value() == 2
+	})
+	assert.NoError(t, err)
+
+	assert.Equal(t, int64(3), rt.GetDatabase().DbStats.Database().SyncFunctionCount.Value())
+
+	var syncData4 db.SyncData
+	_, err = dataStore.GetXattr(ctx, docKey, base.SyncXattrName, &syncData4)
+	assert.NoError(t, err)
+
+	assert.Equal(t, base.Crc32cHashString(updateVal), syncData4.Crc32c)
+	assert.Equal(t, syncData3.Crc32cUserXattr, syncData4.Crc32cUserXattr)
+}
+
+func TestUserXattrOnDemandImportGET(t *testing.T) {
+	if !base.TestUseXattrs() {
+		t.Skip("This test only works with XATTRS enabled")
+	}
+
+	if !base.IsEnterpriseEdition() {
+		t.Skipf("test is EE only - user xattrs")
+	}
+
+	base.SetUpTestLogging(t, base.LevelDebug, base.KeyAll)
+
+	docKey := t.Name()
+	xattrKey := "myXattr"
+	channelName := "testChan"
+
+	// Sync function to set channel access to whatever xattr is
+	rt := rest.NewRestTester(t, &rest.RestTesterConfig{
+		DatabaseConfig: &rest.DatabaseConfig{DbConfig: rest.DbConfig{
+			AutoImport:   false,
+			UserXattrKey: &xattrKey,
+		}},
+		SyncFn: `
+			function (doc, oldDoc, meta){
+				if (meta.xattrs.myXattr !== undefined){
+					channel(meta.xattrs.myXattr);
+				}
+			}`,
+	})
+
+	defer rt.Close()
+
+	dataStore := rt.GetSingleDataStore()
+
+	userXattrStore, ok := base.AsUserXattrStore(dataStore)
+	if !ok {
+		t.Skip("Test requires Couchbase Bucket")
+	}
```

Comment:
this should be always true, can we change this to require.True(t, ok)? If rosmar doesn't implement this, it should.